grpc-protocol/src/main/java/com/linecorp/armeria/common/grpc/protocol/ArmeriaMessageDeframer.java

```diff
@@ -317,6 +328,12 @@ public void closeWhenComplete() {
      */
     @Override
     public void close() {
+        if (whenClosed == null) {
+            whenClosed = CLOSED_FUTURE;
+        } else {
+            whenClosed.doComplete();
```

Comment:
Shouldn't we do this after cleanup the resources? which means at the end of this method?

Reply:
Fixed. :-)